src/parse.ts

```
@@ -188,9 +188,14 @@ export function stringifyParsedURL(parsed: Partial<ParsedURL>): string {
   const proto =
-    parsed.protocol || parsed[protocolRelative]
-      ? (parsed.protocol || "") + "//"
-      : "";
+    // Opaque schemes (the ones parseURL special-cases) carry their body in
+    // pathname and have no authority, so they must not gain a "//" — otherwise
+    // "data:…" round-trips to the invalid "data://…".
+    parsed.protocol && /^(?:blob|data|javascript|vbscript):$/i.test(parsed.protocol)
+      ? parsed.protocol
+      : parsed.protocol || parsed[protocolRelative]
+        ? (parsed.protocol || "") + "//"
+        : "";
```

ESLint

[error] Nested ternary expression should be parenthesized.

(unicorn/no-nested-ternary)